core/http-auth/src/main/java/software/amazon/awssdk/http/auth/internal/DefaultAwsV4HttpSigner.java

```diff
@@ -221,8 +201,72 @@ public AsyncSignedRequest signAsync(AsyncSignRequest<? extends AwsCredentialsIde
     /**
      * Adds session credentials to the request given.
      */
-    private void addSessionCredentials(SdkHttpRequest.Builder requestBuilder,
+    public void addSessionCredentials(SdkHttpRequest.Builder requestBuilder,
                                        AwsSessionCredentialsIdentity credentials) {
         requestBuilder.putHeader(SignerConstant.X_AMZ_SECURITY_TOKEN, credentials.sessionToken());
     }
+
+    private String createContentHash(ContentStreamProvider payload, SdkChecksum checksum) {
+        return HttpChecksumUtils.calculateContentHash(payload, checksum);
+    }
+
+    private CompletableFuture<String> createContentHash(Publisher<ByteBuffer> payload, SdkChecksum checksum) {
+        DigestComputingSubscriber bodyDigester = DigestComputingSubscriber.forSha256(checksum);
+
+        if (payload != null) {
+            payload.subscribe(bodyDigester);
+        }
+
+        return bodyDigester.digestBytes().thenApply(BinaryUtils::toHex);
+    }
+
+    private CanonicalRequestV2 createCanonicalRequest(SignRequest<?, ? extends AwsCredentialsIdentity> signRequest,
+                                                      SdkHttpRequest request,
+                                                      ContentChecksum contentChecksum) {
+        Boolean doubleUrlEncode = validatedProperty(signRequest, DOUBLE_URL_ENCODE, true);
+        Boolean normalizePath = validatedProperty(signRequest, NORMALIZE_PATH, true);
```

Comment:
I thought we had discussed this in earlier PR, but probably missed, these should be `b`ooleans.